src/qt/bitcoingui.cpp

```diff
@@ -1084,6 +1090,22 @@ void BitcoinGUI::setEncryptionStatus(int status)
 }
 #endif // ENABLE_WALLET
 
+void BitcoinGUI::updateProxyIcon()
+{
+    std::string ip_port;
+    bool proxy_enabled = clientModel->getProxyInfo(ip_port);
+
+    if (proxy_enabled) {
+        QString ip_port_q = QString::fromStdString(ip_port);
+
+        labelProxyIcon->setPixmap(platformStyle->SingleColorIcon(":/icons/proxy").pixmap(STATUSBAR_ICONSIZE, STATUSBAR_ICONSIZE));
```

Comment:
If idea is to show/hide then the pixmap can be set when the label is created.

Reply:
In the future, I plan to support a Tor icon, so I prefer to do this here

Reply:
Even in that case it can't change in runtime right?

Reply:
Although toggling proxy requires a restart, I prefer it if the UI doesn't assume that (to prevent a regression later). Not a show-stopper perse, but this code could confuse people.

Suggestion: add an `if` statement that checks if the icon has already been drawn. If so, call `show()`.